### ISSUE TITLE
Function for datetime string between

### ISSUE THREAD

**Issue by ctholho** [supabase-community/copycat#68]
# Feature request

## Is your feature request related to a problem? Please describe.

I need to create dates that are definitely after a specific date (i.e. a date I generated beforehand with copycat). Right now, copycat.dateString only allows us to set a start and end year.

## Describe the solution you'd like

I think it would be a great extension if we can generate datetime strings with a specific start and end time. Same as fakers `date.between()` [see fakerJS docs](https://fakerjs.dev/api/date.html#between)

## Describe alternatives you've considered

In the meantime, a solution could look like this (without proper guards for brevity)

```ts
import { copycat } from '@snaplet/copycat';

type GenerateDateOptions = {
  from: Date;
  to: Date;
};

function dateStringBetween(input: string, options: GenerateDateOptions): Date {
  const fromTimestamp = options.from.getTime();
  const toTimestamp = options.to.getTime();

  const randomTimestamp = copycat.int(input, { min: fromTimestamp, max: toTimestamp });

  return new Date(randomTimestamp).toISOString()
}
```

## Additional context

Would you accept a PR into that direction?

**Comment by justinvdm** [comment]
Hey @ctholho, good point, this sounds like something copycat should have.

I'm not technically an owner of this repo anymore, but I'd think a PR would be welcome.

On how it should look, my take: `dateString` should be extended to take in `min` and `max` options, and changed in fictional (the underlying library where `dateString()` resides): https://github.com/oftherivier/fictional (you're welcome to make a PR there too). Copycat just re-exports dateString, so once fictional is changed (and after the version range for it in copycat is bumped), copycat would have it too.

**Comment by justinvdm** [comment]
@ctholho Thanks to your PR https://github.com/oftherivier/fictional/pull/41, this should now be supported in fictional 2.1.0. 

I just released copycat 5.1.0 with the fictional version range bumped, so that should wrap this up now :)

**Comment by ctholho** [comment]
you're awesome! thank you for the quick support.
